client/core/simnet_trade.go

```diff
@@ -1193,91 +1172,98 @@ func (s *simulationTest) monitorTrackedTrade(client *simulationClient, tracker *
 		client.log.Infof("updated %s balance diff with %s", a.symbol, a.valFmt(amt))
 		client.expectBalanceDiffs[a.id] += amt
 	}
+	for _, match := range tracker.matches {
+		if match.Side == order.Taker {
+			if finalStatus >= order.TakerSwapCast {
+				recordBalanceChanges(true, match.Quantity, match.Rate)
+			}
+			if finalStatus >= order.MatchComplete {
+				recordBalanceChanges(false, match.Quantity, match.Rate)
+			}
+		}
+		if match.Side == order.Maker {
+			if finalStatus >= order.MakerSwapCast {
+				recordBalanceChanges(true, match.Quantity, match.Rate)
+			}
+			if finalStatus >= order.MakerRedeemed {
+				recordBalanceChanges(false, match.Quantity, match.Rate)
+			}
+		}
+	}
+	return nil
```

Comment:
This fixes the btc_dcr balance change issue. Before it was added as the swap progressed, but sometimes it would miss a match status change. We already know what balances to expect before the swap progresses so add it up at the beginning.